### PR TITLE
mzo added from centWave & some polarity issues

### PR DESCRIPTION
**Pull request by hpbenton**
Added polarity splits for MPI function as well - before was only in non parallel version
Added export of mzo & mzoscan from centWave

### CODE REVIEW

File: R/MPI.R [hunk 1]
```diff
@@ -314,12 +314,23 @@ findPeaksPar <- function(arg) {
...
+    		stop("Attempt to select polarity not present in file\n")
+    	}
+    }
     params["object"] <- xRaw
 
     ## remove parameters which are not used by method() from the parameter list
     params["method"] <- params["id"] <- params["profmethod"] <- params["profparam"] <- params["includeMSn"] <- params["lockMassFreq"] <-  params["mslevel"] <- NULL
-
-    params["scanrange"] <- NULL ## avoid filtering scanrange twice, first in xRaw then in findPeaks
+    params["scanrange"] <- params["polarity"] <- NULL ## avoid filtering scanrange twice, first in xRaw then in findPeaks
```
Comment (sneumann) [inline R/MPI.R]: Please leave as two individual assignments, 
make intention clearer.

File: R/MPI.R [hunk 2]
```diff
@@ -348,10 +359,10 @@ fillPeaksChromPar <- function(arg) {
 
   lcraw <- xcmsRaw(arg$file, profmethod=params$prof$method, profstep = 0)
 
-  if(length(params$dataCorrection) > 1){
-    if(params$dataCorrection[i] == 1)
-      lcraw <- stitch(lcraw, AutoLockMass(lcraw))
-  }
+  # if(length(params$dataCorrection) > 1){
```
Comment (sneumann) [inline R/MPI.R]: Why is that removed ? Never worked ? Now done somewhere else ?

File: NAMESPACE [hunk 3]
```diff
@@ -18,7 +18,9 @@ export(
        "xcmsRaw",
        "xcmsSet",
        "xcmsFragments",
-       "xcmsPapply"
+       "xcmsPapply",
+       "ppm",
```
Comment (sneumann) [inline NAMESPACE]: ppm and ppmDev are not part of this pull request.

File: R/MPI.R (hunk 1 above)
Comment (hpbenton) [inline R/MPI.R]: Updated and changed the above lines as well to make it clearer

File: R/MPI.R (hunk 2 above)
Comment (hpbenton) [inline R/MPI.R]: Steffen,

I never really took the time to look at the parallel version of fillPeaks until recently and noticed that the stitch method has some issue in the way you implemented it in the fillPeaks method. Consequently I’ve removed this. There is also some evidence that the Waters cdf files ow take account of this. Further to this in the xcmsonline version I have access to the RAW files themselves which makes this much easier to implement. 
> Now done somewhere else ?
> 

Sort of - I may take the time to implement correctly in parallel fillPeaks.

Paul

File: NAMESPACE (hunk 3 above)
Comment (hpbenton) [inline NAMESPACE]: Do you want a separate pull request for this ??

File: R/MPI.R [hunk 4]
```diff
@@ -314,12 +314,32 @@ findPeaksPar <- function(arg) {
...
+    	}
+    }
     params["object"] <- xRaw
 
     ## remove parameters which are not used by method() from the parameter list
-    params["method"] <- params["id"] <- params["profmethod"] <- params["profparam"] <- params["includeMSn"] <- params["lockMassFreq"] <-  params["mslevel"] <- NULL
-
-    params["scanrange"] <- NULL ## avoid filtering scanrange twice, first in xRaw then in findPeaks
+    params["method"] 		<- NULL
+	arams["lockMassFreq"	<- NULL
```
Comment (sneumann) [inline R/MPI.R]: Hi, this line does not compile. I am currently fixing this, 
but I feel uncomfortable to push code that could not possibly have been run & tested. 
I will push to a different branch and will ask to test more.

**Comment by sneumann** [comment]
Hi,

I fixed the typo in MPI.R, please pull from https://github.com/sneumann/xcms/tree/hpbenton-master
Please check that it now does what you intended.

Can you also please add documentation to avoid that WARNING,
and this time run R CMD check afterwards ? 

````
* checking for missing documentation entries ... WARNING
Undocumented code objects:
  ‘ppm’ ‘ppmDev’
````

Yours,
Steffen

**Comment by sneumann** [comment]
Hi Paul, 
thanks for the pull request. I'll need a moment to digest the 250 commits, 
and also work with @jotsetung to get the github and SVN in sync. Yours, Steffen

**Comment by hpbenton** [comment]
I was messing around with code and different branches just now. Steffen go ahead and ignore the pull and I'll send a diff later that maybe more helpful and easier. I also want to look at the xcms3 changes, exciting stuff.

**Comment by jotsetung** [comment]
amazing! that's a lot of commits! There might indeed be some difficulties merging that, since the *devel* branch has now become the sort of *master* that is in sync with Bioconductor.
